### ISSUE TITLE
Clarify when to use dependencies vs dev-dependencies

### ISSUE THREAD

**Issue by panwauu** [withastro/docs#1435]
As far as I know, it does not really matter whether you install as dev-dependency or dependency. I think we are supposed to install everything as dev-dependency due to SSG? How does this change with SSR?

I think it should be mentioned somewhere. I did not find anything in the reference, not sure about the best place.

There are regularly some support discussions on discord that are related or solved by changing the dependencies:
[Discord: devDependencies over dependencies?](https://discordapp.com/channels/830184174198718474/956662458343366687)
[Discord: Tiny TypeScript question](https://discordapp.com/channels/830184174198718474/1004687477866496030)
[Discord: dev dependency and/or peer dependency](https://discordapp.com/channels/830184174198718474/830184175176122389/971050625162960906)
...

**Comment by sarah11918** [comment]
Thanks for making this issue @panwauu! I agree, I've seen this in support threads too. This sounds like a @Jutanium task to me!  ;)  (Assigning it to Dan, and leaving it to him to un-tag himself if he wants to pass it along.)

**Comment by Jutanium** [comment]
Thanks for assigning this to me! Great timing, too: we just decided to [change the templates](https://github.com/withastro/astro/pull/4544#issuecomment-1231661514) to use `dependencies` only.

I wrote in that PR: 
> No new documentation required, unless we want to prescribe `dependencies` over `devDependencies`

I do want to make it clear in the docs that both choices behave the same. It's up to the user whether they want to split the two (which can be desired if they want to differentiate "build tooling" like Tailwind from "runtime libraries" like React Query).

It seems like there's a bit more discussion to be had on that PR, but as soon as it's approved I'll clarify this point in the docs.
